docs/source/release_notes/release-notes.rst

```diff
@@ -44,6 +44,13 @@ develop
     *    - Type
          - Change
 
+    *    - |new|
+         - Timelock server can now be configured to persist the timestamp bound in the database, specifically in Cassandra/Postgres/Oracle.
+           We recommend this to be configured only for cases where you absolutely need to persist all state in the database, hopefully,
+           in very special cases where backups are simply database dumps and do not have any mechanism for storing timestamps.
+           This will help support large internal product's usage of the Timelock server.
+           (`Pull Request <https://github.com/palantir/atlasdb/pull/2364>`__)
```

Comment:
nit: `hopefully` seems a little awkward here - I'd recommend `for example` instead. Alternatively we can kill off the specific example of a case where it can make sense, if it's likely to confuse users. I'm fine either way.